tests/python_package_test/test_engine.py

```diff
@@ -633,3 +633,25 @@ def is_correctly_constrained(learner):
         }
         constrained_model = lgb.train(params, trainset)
         assert is_correctly_constrained(constrained_model)
+
+    def test_refit(self):
+        X, y = load_breast_cancer(True)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+        X_train, X_valid, y_train, y_valid = train_test_split(X_train, y_train, test_size=0.3, random_state=24)
+        params = {
+            'objective': 'binary',
+            'metric': 'binary_logloss',
+            'verbose': -1,
+            'min_data': 10
+        }
+        lgb_train = lgb.Dataset(X_train, y_train)
+        lgb_eval = lgb.Dataset(X_valid, y_valid, reference=lgb_train)
```

Comment:
I probably was unclear, sorry. I meant, `lgb_eval` are not used in this test (neither for early stopping, nor metric evaluation).

Reply:
oh, got you.